feat(ngUpgrade): simple example

mhevery opened the pull request:
Simple example of how to nest NG1/NG2 components using `ngUpgrade`

@petebacondarwin / @lgalfaso Can you please comment on this PR. The example is completely silly as I could not think of anything else but it demonstrates a working interop.

PascalPrecht commented on modules/examples/src/ngUpgrade/index.html:
```diff
@@ -0,0 +1,22 @@
+    }
+  </style>
+<body>
+  <upgrade-app ng-controller="Index" [user]="name" (reset)="name=''">
```
Can we use controllerAs syntax here?

PascalPrecht commented on modules/examples/src/ngUpgrade/index.ts:
```diff
@@ -0,0 +1,75 @@
+var upgrMod: UpgradeModule = createUpgradeModule();
+
+var ng1module: any = upgrMod.ng1Module;
+ng1module.controller({Index: function($scope) { $scope.name = 'World'; }});
```
As mentioned above, rather use `this` instead of `$scope` with controller as?

mhevery committed:
refactor(ngUpgrade): renames and docs

BREAKING CHANGE:

- Changes the terminology to Adapter and upgrade/downgrade
- Removes the Module from the public API to prevent confusion

mhevery commented:
Closed in favor of: https://github.com/angular/angular/pull/4691
